### PR TITLE
Jsonp configuration

### PR DESCRIPTION
**Pull request by KPP13**
Simple fix: JSONP is returned from the REST controller now.

### CODE REVIEW

File: robot-logic/src/main/java/logic/springconfig/ResetRestController.java [hunk 1]
```diff
@@ -38,4 +42,12 @@ public ResetResponse reset() {
...
     }
+
+
+    @ControllerAdvice
+    public static class JsonpAdvice extends AbstractJsonpResponseBodyAdvice {
+        public JsonpAdvice () {
+            super("callback");
+        }
+    }
 }
```
Comment (kdmrulez) [inline robot-logic/src/main/java/logic/springconfig/ResetRestController.java]: Not clear, what @ControllerAdvice does here.Maybe making a Javadoc for this, will be helpful.

File: robot-logic/src/main/java/logic/springconfig/ResetRestController.java [hunk 2]
```diff
@@ -38,4 +42,12 @@ public ResetResponse reset() {
         LocalDateTime lastDate = TIME_MANAGER.getLastRun();
         return new ResetResponse(lastDate.getHour(), lastDate.getMinute(), lastDate.getSecond(), canRun);
     }
+
+
```
Comment (kdmrulez) [inline robot-logic/src/main/java/logic/springconfig/ResetRestController.java]: Not clear, what @ControllerAdvice does here.Maybe making a Javadoc for it, will be helpful.

File: robot-logic/src/main/java/logic/springconfig/ResetRestController.java [hunk 3]
```diff
@@ -38,4 +42,19 @@ public ResetResponse reset() {
...
+    public static class JsonpAdvice extends AbstractJsonpResponseBodyAdvice {
+
+        /**
+         * This constructor simply invoke parent's constructor to allow using callback.
+         */
+        public JsonpAdvice () {
+            super("callback");
+        }
+    }
 }
```
Comment (danone771909) [inline robot-logic/src/main/java/logic/springconfig/ResetRestController.java:60]: For me it's ok and clear.

Comment (kdmrulez) [reply robot-logic/src/main/java/logic/springconfig/ResetRestController.java:60]: +1

**Pull request merged**